src/map_field.cpp

```diff
@@ -396,6 +396,21 @@ If you need to insert a new field behavior per unit time add a case statement in
 bool map::process_fields_in_submap( submap *const current_submap,
                                     const tripoint &submap )
 {
+    // create all ids once before the loop
+    const field_type_id fd_acid( "fd_acid" );
+    const field_type_id fd_extinguisher( "fd_extinguisher" );
+    const field_type_id fd_fire( "fd_fire" );
+    const field_type_id fd_fungal_haze( "fd_fungal_haze" );
+    const field_type_id fd_fire_vent( "fd_fire_vent" );
+    const field_type_id fd_flame_burst( "fd_flame_burst" );
+    const field_type_id fd_electricity( "fd_electricity" );
+    const field_type_id fd_push_items( "fd_push_items" );
+    const field_type_id fd_shock_vent( "fd_shock_vent" );
+    const field_type_id fd_acid_vent( "fd_acid_vent" );
+    const field_type_id fd_bees( "fd_bees" );
+    const field_type_id fd_incendiary( "fd_incendiary" );
+    const field_type_id fd_fungicidal_gas( "fd_fungicidal_gas" );
```

Comment:
The inner loop here is mostly a case of "do a different thing for each field type".  This could be refactored as a `std::unordered_map<field_type_id, std::function<void()>>` initialized with a bunch of functions, one for each `if` block in the inner loop.  I think you could get away with storing such a map as a `static const` value so it would only need to be constructed once.  (As discussed in other PRs this would be bad if the `int_id`s for these fields changed without restarting CDDA, but I *think* it should be OK for these since they're all core fields; maybe it's a bit too fragile, though...).

It would be safer to use a `std::unordered_map<field_type_str_id, std::function<void()>>`; you'd then have to pay for one `int_id` -> `string_id` conversion per field (which is pretty cheap; just a function call or two and an array lookup) and a string hash conputation.

(You probably actually need to pass some arguments to the functions stored in the map, so it wouldn't actually be a `std::function<void()>`; also you could use a function pointer type instead of `std::function` since I don't think we'd want to use closures here).

Anyway, this is all just brainstorming.  The PR as it stands is already a good step in the right direction.

Reply:
Good point, but I think it's something to do in a future PR, as this one is strictly about performance?

Reply:
I thought something along the lines.  However, I decided to leave it as is.  From the performance standpoint, handful of ifs have negligible performance impact (I think it might be comparable with a single string hash calculation). Your suggestion might be more readable in a long run, especially if the number of cases grows.

BTW, field ids cannot be static, I tried that and game crashes on start.

Reply:
If there are no objections, I'll leave this suggestion for the future research.